### PR TITLE
Improve validation in the Editor component of the Voting Queue modal - Closes #3180

### LINKED ISSUE #3180: Add new validation to the Voting Queue
### Expected behavior
The user should not be allowed to continue if at least one of the voting rows is in the "editing" state. 

### Actual behavior
The user can proceed to the next step. 

### Steps to reproduce
Add a delegate to the voting queue and try proceeding to the next step with the form in the "editing" state.

### PR DESCRIPTION
**Pull request by UsamaHameed**
### What was the problem?
This PR resolves #3180

### How was it solved?
Added new validation to the votes state in the editor component in the voting queue modal

### How was it tested?
Unit tests and manually

### CODE REVIEW

**Review by isalga — APPROVED** [review]

File: src/components/screens/votingQueue/editor/editor.js [hunk 1]
```diff
@@ -131,7 +132,10 @@ const Editor = ({
 
   const { added, edited, removed } = useMemo(() => getVoteStats(votes), [votes]);
   const feedback = validateVotes(votes, account.balance, fee.value, t);
-  const isCTADisabled = feedback.error || Object.keys(changedVotes).length === 0;
+  const areVotesInValid = Object.values(votes).some(vote =>
+    (vote.unconfirmed === '' || vote.unconfirmed === undefined));
```
Comment (reyraa) [inline src/components/screens/votingQueue/editor/editor.js:136]: I think we should've done this in the validator function, alongside the other checks.

File: src/components/screens/votingQueue/editor/editor.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ const getVoteStats = votes =>
 const token = tokenMap.LSK.key;
 const txType = 'vote';
 
+/* eslint-disable max-statements */
```
Comment (reyraa) [inline src/components/screens/votingQueue/editor/editor.js:105]: Handling a simple form state shouldn't be so complex. we have this problem in Send form too. some of it due to the complicated usage of `useTransactionFeeCalculation`. We should consider rethinking it later on.

Comment (UsamaHameed) [reply src/components/screens/votingQueue/editor/editor.js:105]: We can separate it out to yet another hooook 😄

File: src/components/screens/votingQueue/editor/editor.js (hunk 1 above)
Comment (UsamaHameed) [reply src/components/screens/votingQueue/editor/editor.js]: I have moved this to the validateVotes function

**Review by isalga — APPROVED** [review]

File: src/components/screens/votingQueue/editor/editor.js (hunk 2 above)
Comment (reyraa) [reply src/components/screens/votingQueue/editor/editor.js]: LOL

**Review by reyraa — APPROVED** [review]

**Pull request merged**